bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -1105,6 +1109,18 @@ public void updateVolume() {
         }
     }
 
+    public void updateCurrentProfile() {
+        if (socket.isConnected()) {
+            JsonElement response = socket.callMethod("Profiles.GetCurrentProfile");
+
+            try {
+                listener.updateCurrentProfile(gson.fromJson(response, KodiProfile.class).getLabel());
+            } catch (JsonSyntaxException e) {
+                logger.error("Json syntax exception occurred: {}", e.getMessage(), e);
```

Comment:
Shouldn't this be log to debug? Same for other log to error.

Reply:
I'd say, if the JSON syntax is not as expected, it is a bug in the binding (possibly because Kodi changed its format) and thus users should see and report it.

Reply:
It make sense to report this kind of errors but our general policy is to only use error log for system wide failures. At least that is how we apply that in reviews in general.Or is that to strict?

Reply:
Hm, it seems that somebody changed the logging guidelines since I read them. What I remembered was [this](https://www.eclipse.org/smarthome/documentation/development/guidelines.html#e-logging):

> error logging [...] should also be used if some code fails irrecoverably and the user should report it as a severe bug.

In our [latest guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging), it seems to be slightly contradictory as it has those statements:

> Exceptions with stacktraces in the log are considered to be bugs in your binding that should be reported and fixed.

-> This means for me that it must not be DEBUG level as otherwise the regular user would not see the message.

> error logging is not allowed in extensions and is purely reserved to the framework. The only exception would be if something is going really, really wrong in your extension, and there is a possibility that the stability of the framework could be affected.

This does not allow reporting severe bugs in an add-on itself. Imho this statement is too strong.

Reply:
I usually recommend logging at WARN if it should be reported. The first one should be removed. Sometimes the stacktrace helps debugging (e.g. IOException connection issues) but sometimes the same exception is just a temporary issue and will heal by itself.